src/bio2bel/sources/intact.py

```diff
@@ -0,0 +1,633 @@
+        with zip_file.open('intact.txt') as file:
+            df = pd.read_csv(file, sep='\t', usecols=COLUMNS, na_values={'-'})
+
+    df.dropna(inplace=True)
+
+    # Omit certain interaction types
+    df = df[~df['Interaction type(s)'].isin(INTACT_OMIT_INTERACTIONS)]
+
+    df['#ID(s) interactor A'] = df['#ID(s) interactor A'].map(_process_interactor)
+    df['ID(s) interactor B'] = df['ID(s) interactor B'].map(_process_interactor)
+
+    # filter for pubmed
+    logger.info('mapping provenance')
+    df['Publication Identifier(s)'] = df['Publication Identifier(s)'].map(_process_pmid)
+
+    # filter for intact-miscore
+    df['Confidence value(s)'] = df['Publication Identifier(s)'].map(_process_score)
+
+    return df
+
+
+def get_bel() -> BELGraph:
+    """Get BEL graph."""
+    df = get_processed_intact_df()
+    graph = BELGraph(name=MODULE_NAME, version=VERSION)
+    it = tqdm(df[COLUMNS].values, total=len(df.index), desc=f'mapping {MODULE_NAME}', unit_scale=True)
+    for source_uniprot_id, target_uniprot_id, relation, pubmed_id, detection_method, source_db, confidence in it:
+        if pd.isna(source_uniprot_id) or pd.isna(target_uniprot_id):
+            continue
+
+        _add_row(
+            graph,
+            relation=relation,
+            source_uniprot_id=source_uniprot_id,
+            target_uniprot_id=target_uniprot_id,
+            pubmed_id=pubmed_id,
+            int_detection_method=detection_method,
+            source_database=source_db,
+            confidence=confidence,
+        )
+    return graph
+
+
+def _add_row(
+    graph: BELGraph,
+    relation: str,
+    source_uniprot_id: str,
+    target_uniprot_id: str,
+    pubmed_id: str,
+    int_detection_method: str,
+    source_database: str,
+    confidence: str,
+) -> None:  # noqa:C901
+    """Add for every pubmed ID an edge with information about relationship type, source and target.
+
+    :param graph: graph to add edges to
+    :param relation: row value of column relation
+    :param source_uniprot_id: row value of column source
+    :param target_uniprot_id: row value of column target
+    :param pubmed_id: row value of column pubmed_id
+    :param int_detection_method: row value of column interaction detection method
+    :param confidence: row value of confidence score column
+    :return: None
+    """
+    if pubmed_id is None:
+        return
+    print(confidence)
+    annotations = {
+        'psi-mi': relation,
+        'intact-detection': int_detection_method,
+        'intact-source': source_database,
+        'intact-confidence': confidence,
+    }
+    # map double spaces to single spaces in relation string
+    relation = ' '.join(relation.split())  # FIXME how often does this happen? can you tweet Intact with the number?
+    # I only found it in 'psi-mi:"MI:1237"(proline isomerization reaction)', nowhere else
+    source = pybel.dsl.Protein(
+        namespace='uniprot',
+        identifier=source_uniprot_id[1],
```

Comment:
you should rename this variable. it's a pair of prefix-identifier. This means you should save them into two different variables
`source_prefix, source_id = ...`

Then you can use if statements to decide based on the source prefix how you should handle the entities, in addition to all of the other logic for handling the relationships.

Reply:
e.g., what if you get something that isn't actually a uniprot id? don't throw that info away

Reply:
so should I just add the protein, no matter which identifier it has? I'm not really comfortable with this

Reply:
Also, you put in this counter: 
```python
 Counter({'chebi': 9534,
         'ensembl': 3156,
         'refseq': 444,
         'ensemblgenomes': 439,
         'ddbj/embl/genbank': 204,
         'wwpdb': 163,
         'matrixdb': 102,
         'reactome': 87,
         'intenz': 43,
         'signor': 15,
         'chembl target': 11,
         'dip': 4,
         'entrezgene/locuslink': 2,
         'protein ontology': 2,
         'emdb': 2})

```
Would you want to include all of them?

Reply:
No what I meant is you should be checking the namespaces and adding different kind of relations based on them. Or filter so you only keep uniprot ahead of time. Either one is fine, but currently you’re just overwriting the namespace for everything to uniprit